### PR TITLE
chore: add release candidate workflow

### PR DESCRIPTION
**Pull request by Diiaablo95**
Fixes https://github.com/KILTprotocol/ticket/issues/1858 and fixes https://github.com/KILTprotocol/ticket/issues/1877.

On the side, it also deletes seemingly outdated dispatched upon dev releases, which are always failing anyway.

### CODE REVIEW

File: .github/workflows/devpackpublish.yml [hunk 1]
```diff
@@ -48,23 +48,4 @@ jobs:
           git config --local user.email "action@github.com"
           git add .
           git commit -m "[CI Skip] ci: publish prerelease" -n
-          git push
-  dispatch:
```
Comment (Diiaablo95) [inline .github/workflows/devpackpublish.yml:52]: Is it ok to remove these outdated triggers?

Comment (tjwelde) [reply .github/workflows/devpackpublish.yml:52]: yes

File: .github/workflows/devpackpublish.yml [hunk 2]
```diff
@@ -33,7 +33,7 @@ jobs:
       - name: Set all other packages to same version
         env:
           PACKAGE_VERSION: ${{ steps.package_version.outputs.package_version }}
-        run: yarn workspaces foreach -pt --no-private exec npm version --no-git-tag-version $PACKAGE_VERSION
```
Comment (tjwelde) [inline .github/workflows/devpackpublish.yml:36]: That would also set the version for the root package, what has been done right before. Not sure if that is a problem, though.

**Review by tjwelde — APPROVED** [review]
LGTM

File: .github/workflows/devpackpublish.yml (hunk 2 above)
Comment (Diiaablo95) [reply .github/workflows/devpackpublish.yml:36]: The `set:version` does exactly the same as this line that is being replaced, including changing the root package version. Why would this be a problem?

**Review by rflechtner — APPROVED** [review]
Works for me!

**Pull request merged**